src/test/unit/controllers/SubscriptionCaseSearchResultController.test.ts

```diff
@@ -0,0 +1,93 @@
+import sinon from 'sinon';
+import { Response } from 'express';
+import SubscriptionCaseSearchResultController from '../../../main/controllers/SubscriptionCaseSearchResultController';
+import fs from 'fs';
+import path from 'path';
+import {mockRequest} from '../mocks/mockRequest';
+import {HearingRequests} from '../../../main/resources/requests/hearingRequests';
+
+const subscriptionCaseSearchResultController = new SubscriptionCaseSearchResultController();
+const rawData = fs.readFileSync(path.resolve(__dirname, '../mocks/subscriptionCaseList.json'), 'utf-8');
+const subscriptionsCaseData = JSON.parse(rawData);
+const stub = sinon.stub(HearingRequests.prototype, 'getSubscriptionCaseDetails');
+stub.withArgs('ABC12345').returns(subscriptionsCaseData);
+
+describe('Subscription Search Case Reference Result Controller', () => {
+  let i18n = {};
+  it('should render the search result page', () => {
+
+    i18n = {
+      'subscription-case-search': {},
+    };
+
+    const response = {
+      render: function () {
+        return '';
+      },
+    } as unknown as Response;
+    const request = mockRequest(i18n);
+    request.query = { 'search-input': 'ABC12345'};
+    const responseMock = sinon.mock(response);
+
+    const expectedData = {
+      ...i18n['subscription-search-case-results'],
+      searchInput : 'ABC12345',
+      searchResults: subscriptionsCaseData,
+    };
+
+    responseMock.expects('render').once().withArgs('subscription-search-case-results', expectedData);
+
+    return subscriptionCaseSearchResultController.get(request, response).then(() => {
+      responseMock.verify();
+    });
+  });
+
+  it('should render an error page if search input does not return any results', () => {
+
+    stub.withArgs('12345678').returns(null);
+
+    const i18n = {
+      'subscription-search-case-results': {},
+    };
+
+    const response = {
+      render: function() {return '';},
+    } as unknown as Response;
+
+    const request = mockRequest(i18n);
+    request.query = {'search-input': ''};
+
+    const responseMock = sinon.mock(response);
+
+    const expectedData = {
+      ...i18n['error'],
+    };
+
+    responseMock.expects('render').once().withArgs('error', expectedData);
+
+    return subscriptionCaseSearchResultController.get(request, response).then(() => {
+      responseMock.verify();
+    });
+  });
+
+  it('should redirect to pending subscription page when user click on Continue', () => {
+
+    const response = {
+      redirect: function() {return '';},
+      render: function() {return '';},
```

Comment:
if you include the response with both method calls in then you can act on the comment maro made above about moving response delcaration to class level rather then test level

Reply:
fixed.